client/me/happychat/main.jsx

```diff
@@ -29,25 +43,102 @@ export class HappychatPage extends Component {
 	}
 
 	render() {
+		const {
+			chatStatus,
+			connectionStatus,
+			currentUserEmail,
+			disabled,
+			isCurrentUser,
+			isExternalUrl,
+			isServerReachable,
+			message,
+			onSendMessage,
+			onSendNotTyping,
+			onSendTyping,
+			onSetCurrentMessage,
+			timeline,
+			translate,
+			twemojiUrl,
+		} = this.props;
+
 		return (
 			<div className="happychat__page" aria-live="polite" aria-relevant="additions">
 				<HappychatConnection />
-				<Timeline />
-				<Notices />
-				<Composer />
+				<Timeline
+					currentUserEmail={ currentUserEmail }
+					isCurrentUser={ isCurrentUser }
+					isExternalUrl={ isExternalUrl }
+					timeline={ timeline }
+					translate={ translate }
+					twemojiUrl={ twemojiUrl }
+				/>
+				<Notices
+					chatStatus={ chatStatus }
+					connectionStatus={ connectionStatus }
+					isServerReachable={ isServerReachable }
+					translate={ translate }
+				/>
+				<Composer
+					disabled={ disabled }
+					message={ message }
+					onSendMessage={ onSendMessage }
+					onSendNotTyping={ onSendNotTyping }
+					onSendTyping={ onSendTyping }
+					onSetCurrentMessage={ onSetCurrentMessage }
+					translate={ translate }
+				/>
 			</div>
 		);
 	}
 }
 
 HappychatPage.propTypes = {
+	chatStatus: PropTypes.string,
+	connectionStatus: PropTypes.string,
+	currentUserEmail: PropTypes.string,
+	disabled: PropTypes.bool,
+	isCurrentUser: PropTypes.func,
+	isExternalUrl: PropTypes.func,
+	isServerReachable: PropTypes.bool,
+	message: PropTypes.string,
+	onSendMessage: PropTypes.func,
+	onSendNotTyping: PropTypes.func,
+	onSendTyping: PropTypes.func,
+	onSetCurrentMessage: PropTypes.func,
 	setBlurred: PropTypes.func,
 	setFocused: PropTypes.func,
+	timeline: PropTypes.array,
+	translate: PropTypes.func,
+	twemojiUrl: PropTypes.string,
+};
+
+const isCurrentUserHelper = currentUser => ( { user_id, source } ) => {
+	return user_id.toString() === currentUser.ID.toString() && source === 'customer';
+};
+
+const mapState = state => {
+	const currentUser = getCurrentUser( state );
+	return {
+		chatStatus: getHappychatChatStatus( state ),
+		connectionStatus: getHappychatConnectionStatus( state ),
+		currentUserEmail: currentUser.email,
+		disabled: ! canUserSendMessages( state ),
+		isCurrentUser: isCurrentUserHelper( currentUser ), // see redux-no-bound-selectors eslint-rule
```

Comment:
The same opinion as above :)

Reply:
Fixed in 6902e72881ca01dedd3ec88a6289b43ae19acc98

Reply:
Fixed in 6902e72881ca01dedd3ec88a6289b43ae19acc98

Reply:
Fixed in 6902e72